jsonschema/schema.go

```diff
@@ -80,16 +83,16 @@ type Schema struct {
 	Pattern          string   `json:"pattern,omitempty"`
 
 	// arrays
-	PrefixItems      []*Schema `json:"prefixItems,omitempty"`
-	Items            *Schema   `json:"items,omitempty"`
-	MinItems         *int      `json:"minItems,omitempty"`
-	MaxItems         *int      `json:"maxItems,omitempty"`
-	AdditionalItems  *Schema   `json:"additionalItems,omitempty"`
-	UniqueItems      bool      `json:"uniqueItems,omitempty"`
-	Contains         *Schema   `json:"contains,omitempty"`
-	MinContains      *int      `json:"minContains,omitempty"` // *int, not int: default is 1, not 0
-	MaxContains      *int      `json:"maxContains,omitempty"`
-	UnevaluatedItems *Schema   `json:"unevaluatedItems,omitempty"`
+	PrefixItems      []*Schema            `json:"prefixItems,omitempty"`
+	Items            *SchemaOrSchemaArray `json:"items,omitempty"`
```

Comment:
I'd like to avoid introducing those union types. They are hard to work with. I'd like to do here what I did with type and types. Let's keep Items as a schema, and have another field, ItemsArray, that is a []*Schema. For serialization, you can copy what I did with type and types.